pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -334,7 +337,12 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream discovery.AggregatedD
 					return err
 				}
 			default:
-				adsLog.Warnf("ADS: Unknown watched resources %s", discReq.String())
+				// Allow custom generators to work without 'generator' metadata.
+				// It would be an error/warn for normal XDS - so nothing to lose.
+				err = s.handleCustomGenerator(con, discReq)
+				if err != nil {
+					adsLog.Warnf("ADS: Unknown watched resources %s", discReq.String())
```

Comment:
This error is not really accurate, it could fail for a number of reasons - and we don't actually report the error here.

Reply:
Fixed - now API gen will log 'unknown watched resource', and if it fails - we return (as on all other paths)